r4/datatypes_primitive_types.bal

```diff
@@ -89,14 +108,20 @@ public type time string;
 # whitespace, and where there is no whitespace other than single spaces in the contents
 # * Regex: `[^\s]+(\s[^\s]+)*`
 @constraint:String {
-    pattern: re `[^\s]+(\s[^\s]+)*`
+    pattern: {
+        value: re `[^\s]+(\s[^\s]+)*`,
+        message: "A code is restricted to a string with at least one character and no leading or trailing whitespace, and there is no whitespace other than single spaces in the contents."
+    }
 }
 public type code string;
 
 # An OID represented as a URI (RFC 3001 ); e.g. urn:oid:1.2.3.4.5
 # * Regex: `urn:oid:[0-2](\.(0|[1-9][0-9]*))+`
 @constraint:String {
-    pattern: re `urn:oid:[0-2](\.(0|[1-9][0-9]*))+`
+    pattern: {
+        value: re `urn:oid:[0-2](\.(0|[1-9][0-9]*))+`,
+        message: "An OID represented as a URI (RFC 3001 ); e.g. urn:oid:1.2.3.4.5"
```

Comment:
Shouldn't the validation failed message would be something like below, the current one seems to be explaining the data type.

"This field should be of 'string' type. Refer https://hl7.org/fhir/R4/datatypes.html#string for more info."

Reply:
We can't use "this field" because these are primitive data types not any fields in resources.

Reply:
Fixed: https://github.com/ballerina-platform/module-ballerinax-health.fhir.r4/pull/426/commits/758d077505f71c274be73c4619685d5429e80be3